src/declarations/stencil-public-compiler.ts

```diff
@@ -2233,8 +2234,15 @@ export interface OutputTargetHydrate extends OutputTargetBase {
 export interface OutputTargetCustom extends OutputTargetBase {
   type: 'custom';
   name: string;
+  /**
+   * Indicate in wich mode the output target has to be executed.
+   * By default if nothing is specified it run for both.
+   * - `"run"`: Executed in dev mode on every change in your code
+   * - `"build"`: Executed only on build
+   */
+  task?: 'run' | 'build';
```

Comment:
For the comment, just a few wording things, and I'm wondering if the variants here and the configuration name should change? Just brainstorming but I'm thinking something like this:

```suggestion
  /**
   * Indicate when the output target should be executed.
   * 
   * - `"onBuildOnly"`: Executed only when `stencil build` is called without `--watch`.
   * - `"always"`: Executed on every build, including in `watch` mode.
   *
   * Defaults to "always".
   */
  taskShouldRun?: 'onBuildOnly' | 'always';
```